### PR TITLE
Tweak D for C, C++ programmer pages

### PR DESCRIPTION
**Pull request by ntrel**
Also expand factorial example.

### CODE REVIEW

File: ctod.dd [hunk 1]
```diff
@@ -582,7 +582,7 @@ $(H3 <a name="tagspace">Struct tag name space</a>)
 
 $(H4 The C Way)
 
-       It's annoying to have to put the struct keyword every time a type is specified,
+       It's annoying to have to put the `struct` keyword every time a type is specified,
```
Comment (Hackerpilot) [inline ctod.dd]: Also replace "put" with "use" or "write".

File: ctod.dd [hunk 2]
```diff
@@ -295,7 +295,7 @@ for (i = 0; i &lt; sizeof(array) / sizeof(array[0]); i++)
 
 $(H4 The D Way)
 
-The length of an array is accessible through the property "length".
+The length of an array is accessible through the property `length`.
```
Comment (Hackerpilot) [inline ctod.dd]: "`length` property"

File: cpptod.dd [hunk 3]
```diff
@@ -800,7 +809,7 @@ template IsFunctionT(T)
 
 void test()
 {
-    typedef int fp(int);
+    alias int fp(int);
```
Comment (JackStouffer) [inline cpptod.dd]: new alias syntax please

Comment (ntrel) [inline cpptod.dd]: Not sure how, I tried `alias fp = int function(int)` but that didn't work.

**Comment by ntrel** [comment]
@Hackerpilot done

File: cpptod.dd (hunk 3 above)
Comment (Hackerpilot) [inline cpptod.dd]: I think I've run into this issue before. If it is what I think it is, it's not possible to translate this to the new alias syntax.

**Pull request merged**